src/util/RtreeBuild.cpp

```diff
@@ -0,0 +1,1133 @@
+   */
+  bool lookingForX = true;
+  bool readingDouble = false;
+  std::string currentDouble;
+
+  double minX = -1;
+  double maxX = -1;
+  double minY = -1;
+  double maxY = -1;
+
+  for (char c : wkt) {
+    if (isdigit(c)) {
+      readingDouble = true;
+      currentDouble += c;
+    } else if (c == '.') {
+      readingDouble = true;
+      currentDouble += '.';
+    } else if (c == ' ') {
+      if (readingDouble && lookingForX) {
+        // x is completely read in
+        readingDouble = false;
+        lookingForX = false;
+        double x;
+        try {
+          x = std::stod(currentDouble);
+        } catch(...) {
+          return { };
+        }
+        currentDouble = "";
+        if (x < minX || minX == -1) {
+          minX = x;
+        }
+
+        if (x > maxX) {
+          maxX = x;
+        }
+      }
+    } else {
+      if (readingDouble && !lookingForX) {
+        // y is completely read in
+        readingDouble = false;
+        lookingForX = true;
+        double y;
+        try {
+          y = std::stod(currentDouble);
+        } catch(...) {
+          return { };
+        }
+        currentDouble = "";
+        if (y < minY || minY == -1) {
+          minY = y;
+        }
+
+        if (y > maxY) {
+          maxY = y;
+        }
+      }
+    }
+  }
+
+  return { Rtree::createBoundingBox(minX, minY, maxX, maxY) };
+}
+
+std::optional<Rtree::BoundingBox> Rtree::ConvertWordToRtreeEntry(const std::string& wkt) {
+  /**
+     * Convert a single wkt literal to a boundingbox.
+   */
+  std::optional<Rtree::BoundingBox> boundingBox;
+
+  /* Get the bounding box(es) of either a multipolygon, polygon or a linestring */
+  std::size_t posWKTStart = wkt.find("MULTIPOLYGON(((") + 14;
+  std::size_t posWKTEnd = wkt.find(")))", posWKTStart);
+  if (posWKTStart != std::string::npos && posWKTEnd != std::string::npos) {
+    std::string newWkt = wkt.substr(posWKTStart, posWKTEnd - posWKTStart + 1);
+    boundingBox = GetBoundingBoxFromWKT(newWkt);
+  } else {
+    posWKTStart = wkt.find("POLYGON((") + 8;
+    posWKTEnd = wkt.find("))", posWKTStart);
+    if (posWKTStart != std::string::npos && posWKTEnd != std::string::npos) {
+      std::string newWkt = wkt.substr(posWKTStart, posWKTEnd - posWKTStart + 1);
+      boundingBox = GetBoundingBoxFromWKT(newWkt);
+    } else {
+      posWKTStart = wkt.find("LINESTRING(") + 10;
+      posWKTEnd = wkt.find(')', posWKTStart);
+      if (posWKTStart != std::string::npos && posWKTEnd != std::string::npos) {
+        std::string newWkt = wkt.substr(posWKTStart, posWKTEnd - posWKTStart + 1);
+        boundingBox = GetBoundingBoxFromWKT(newWkt);
+      } else {
+        return { };
+      }
+    }
+  }
+
+  return boundingBox;
+}
+
+void Rtree::SaveEntry(Rtree::BoundingBox boundingBox, uint64_t index, std::ofstream& convertOfs) {
+  /**
+     * Save a single entry (which was e.g. converted by ConvertWordToRtreeEntry) to the disk
+   */
```

Comment:
doesn't just
```
static_assert(std::is_trivially_copyable_v<Rtree::BoundingBox>;
ofs.write(reinterpret_cast<const char *>(&boundingBox, sizeof(boundingBox));
```
do the write thing?

(The same for the `WithOrderIndex` variation below.)

Reply:
It does. And this writing should be in the same file as the `FileReader`.